src/network/networkd-link.h

```diff
@@ -92,6 +92,7 @@ typedef struct Link {
         Set *addresses_foreign;
         Set *routes;
         Set *routes_foreign;
+        Set *rules;
```

Comment:
hmm, maybe I missing something, but this Set is never freed, is it?

Reply:
Yes getting freed at `manager_free`.

Reply:
manager_free frees this field of the Link structure? Not following? manager_free should free the field of the Manager structure, and link_free() the one of the Link structure, no?

Reply:
actually rules are not bound to any of the interface . So these are kept in manager.

Reply:
but this Set* is part of the "Link" structure anyway?

Reply:
oops it's not used actually . Sorry for the confusion. updated